code/game/dna/dna2.dm

```diff
@@ -92,7 +93,8 @@ var/global/list/datum/dna/gene/dna_genes[0]
 	new_dna.b_type=b_type
 	new_dna.real_name=real_name
 	new_dna.species=species
-	new_dna.body_markings=body_markings.Copy()
+	var/list/body_markings_genetic = (body_markings - body_marking_nopersist_list)
```

Comment:
Are body markings an associative list by name? That... seems a little overcomplicated to me, but it does simplify logic here.